bundles/org.openhab.binding.wizlighting/src/main/java/org/openhab/binding/wizlighting/internal/utils/WizResponseDeserializer.java

```diff
@@ -0,0 +1,184 @@
+        } else if (context == null) {
+            logger.trace("No context available for parsing sub-objects.");
+        } else {
+            jobject = json.getAsJsonObject();
+
+            // Parse the ID
+            deserializedResponse.setId(jobject.get("id").getAsInt());
+            // Parse the environment - I think this is always sent, but I'm checking anyway
+            if (jobject.has("env")) {
+                deserializedResponse.setEnv(jobject.get("env").getAsString());
+            }
+
+            // Check if the response contains an error
+            // Return without completing parsing if there's an error
+            if (jobject.has("error")) {
+                ErrorResponseResult error = context.deserialize(jobject.getAsJsonObject("error"),
+                        ErrorResponseResult.class);
+                deserializedResponse.setError(error);
+                logger.warn("Bulb returned an error on method {}:  {}, {}", jobject.get("method"), error.code,
+                        error.message);
+                return deserializedResponse;
+            }
+
+            // Parse the method. We will use the method to decide how to continue to parse
+            // Bail out of everything if we cannot understand the method.
+            WizLightingMethodType method;
+            if (jobject.has("method")) {
+                try {
+                    String inMethod = jobject.get("method").getAsString();
+                    method = WizLightingMethodType.valueOf(inMethod);
+                    deserializedResponse.setMethod(method);
+                } catch (IllegalArgumentException e) {
+                    logger.warn("Bulb returned an invalid method: {}", jobject.get("method"));
+                    return deserializedResponse;
+                }
+            } else {
+                throw new JsonParseException("Incoming message did not contain a method and cannot be parsed!");
+            }
+
+            switch (method) {
+                case registration:
+                    // {"method": "registration", "id": 1, "env": "pro", "result": {"mac":
+                    // "macOfopenHAB", "success": true}}
+                    logger.trace("Deserializing registration result");
+                    JsonObject registrationResult = jobject.getAsJsonObject("result");
+                    String mac = registrationResult.get("mac").getAsString();
+                    deserializedResponse.setWizResponseMacAddress(mac);
+                    deserializedResponse.setResultSucess(registrationResult.get("success").getAsBoolean());
+                    logger.trace("Registration result deserialized with mac {} and success {}", mac,
+                            registrationResult.get("success").getAsBoolean());
+                    break;
+
+                case pulse:
+                    // {"method":"pulse","id":22,"env":"pro","result":{"success":true}}
+                case setPilot:
```

Comment:
Is pulse meant to fall through to setPilot?

Reply:
yes